app/views/search/_form.html.haml

```diff
@@ -33,3 +42,32 @@
 
       return false;
     });
+
+    $('#new_search').submit(function() {
+      var emptyCityGeoCount = 4;
+      $(this).find('input.hidden').each(function() {
+        var value = $(this).val();
+        if (value && value !== '') {
+          emptyCityGeoCount -= 1;
+        }
+      });
+      if (emptyCityGeoCount > 2) { // No city fully selected with geo
+        var input = $('#search_to_city');
+
+        // Display error
+        input.parent().addClass('has-error');
+        var inputHelpBlock = input.next('.help-block');
+        if (inputHelpBlock.length > 0) {
+          inputHelpBlock.show();
+        } else { // Create it
+          $("<span>", {
```

Comment:
As you said it, this looks too ugly for deployment. Could avoid repetition with l 47 and l 50 of search.rb (same message in 3 places)

Reply:
True ! I will clean this up.